### PR TITLE
Refactor distributed dtensor modules

### PR DESCRIPTION
**Pull request by AllenYolk**
## Summary
- split the distributed dtensor implementation into focused analysis/config/execution/mesh/runtime/FSDP/optimizer/TP/pipeline modules while keeping the legacy dtensor facade
- add shared eager config assembly for API, adapters, training, and benchmark paths
- reorganize distributed tests by module and rewrite the distributed tutorials/API page; updated the external ObsidianVault architecture note

## Validation
- PYTHONPATH="/Users/allenyolk/.codex/worktrees/0e0c/spikingjelly-dev" /Users/allenyolk/CodeRepo/spikingjelly-dev/.venv/bin/python -m pytest test/activation_based/test_distributed*.py test/activation_based/test_train_distributed_v2.py test/activation_based/test_benchmark_snn_distributed.py -q
- uvx ruff check spikingjelly/activation_based/distributed spikingjelly/activation_based/examples/memopt/train_distributed.py benchmark/benchmark_snn_distributed.py test/activation_based/test_distributed_dtensor.py test/activation_based/test_distributed_*.py test/activation_based/test_train_distributed_v2.py test/activation_based/test_benchmark_snn_distributed.py
- git diff --check
- PYTHONPATH="/Users/allenyolk/.codex/worktrees/0e0c/spikingjelly-dev" /Users/allenyolk/CodeRepo/spikingjelly-dev/.venv/bin/sphinx-build -W --keep-going -b html docs/source docs/build/html
- g3 GPU smoke, synced worktree at commit 1e73ffcc, with CUDA_DEVICE_ORDER=PCI_BUS_ID NCCL_P2P_DISABLE=1 NCCL_IB_DISABLE=1, avoiding physical GPU 2:
  - 2 GPU CIFAR10-DVS VGG TP: passed, /tmp/sj-dist-cifar_tp.log
  - 2 GPU CIFAR10-DVS VGG FSDP2: passed, /tmp/sj-dist-cifar_fsdp2.log
  - 4 GPU CIFAR10-DVS VGG FSDP2+TP mesh 2x2: passed, /tmp/sj-dist-cifar_fsdp2_tp.log
  - 2 GPU CIFAR10-DVS VGG PP/1F1B: passed, /tmp/sj-dist-cifar_pp_1f1b.log
  - 2 GPU Spikformer TP small image: passed, /tmp/sj-dist-spikformer_tp.log

## Summary by CodeRabbit

* **New Features**
  * Expanded distributed **Analyze → Plan → Apply** workflow with eager-policy-driven **dp/tp/fsdp2/fsdp2_tp** configuration and improved auto-strategy recommendations.
  * Added distributed **pipeline** builders for **CIFAR10-DVS VGG** and **Spikformer**, including stage **memopt** selection and schedule/layout validation.
  * Introduced tensor-parallel **channel-sharded Conv/BatchNorm**, tensor-sharded memory module factory, and TP communication debug statistics.
* **Documentation**
  * Reworked English + Chinese distributed training tutorials and API docs (including benchmark logging guidance).
* **Bug Fixes**
  * Hardened benchmark output capture and aligned benchmark metric/JSON record fields.
* **Breaking Changes**
  * Replaced `TensorShardMemoryModule` with `make_tensor_shard_memory_module` (deprecated alias preserved).
* **Tests**
  * Large suite of new/updated distributed analysis, execution, tensor-parallel, pipeline, mesh, and benchmark tests.

### CODE REVIEW

**Comment by AllenYolk** [comment]
@coderabbitai review

**Comment by AllenYolk** [comment]
@gemini-code-assist review

**Comment by AllenYolk** [comment]
@coderabbitai review

**Comment by AllenYolk** [comment]
@gemini-code-assist review

**Comment by AllenYolk** [comment]
@coderabbitai review

**Comment by AllenYolk** [comment]
@gemini-code-assist review

**Comment by AllenYolk** [comment]
@coderabbitai review

**Comment by AllenYolk** [comment]
@gemini-code-assist review

**Comment by AllenYolk** [comment]
@coderabbitai review

**Comment by AllenYolk** [comment]
@gemini-code-assist review

**Comment by AllenYolk** [comment]
@gemini-code-assist review

**Comment by AllenYolk** [comment]
@coderabbitai review

File: spikingjelly/activation_based/distributed/tensor_parallel/channel.py [hunk 1]
```diff
@@ -0,0 +1,581 @@
+import torch
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/tensor_parallel/channel.py:1]: Thanks for the review. We are not targeting Python 3.8/3.9 in this PR: `pyproject.toml` declares `requires-python = ">=3.11"`. Because of that, PEP 585 built-in generics and Python 3.10+ features such as `zip(..., strict=True)` are within the supported runtime, so we will keep this unchanged.

File: spikingjelly/activation_based/distributed/execution.py [hunk 2]
```diff
@@ -0,0 +1,340 @@
+import os
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/execution.py]: Thanks for the review. We are not targeting Python 3.8/3.9 in this PR: `pyproject.toml` declares `requires-python = ">=3.11"`. Because of that, PEP 585 built-in generics and Python 3.10+ features such as `zip(..., strict=True)` are within the supported runtime, so we will keep this unchanged.

File: spikingjelly/activation_based/distributed/pipeline/cifar10dvs_vgg.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
+from typing import Optional, Sequence, Union
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/pipeline/cifar10dvs_vgg.py:1]: Thanks for the review. We are not targeting Python 3.8/3.9 in this PR: `pyproject.toml` declares `requires-python = ">=3.11"`. Because of that, PEP 585 built-in generics and Python 3.10+ features such as `zip(..., strict=True)` are within the supported runtime, so we will keep this unchanged.

File: spikingjelly/activation_based/distributed/pipeline/runtime.py [hunk 4]
```diff
@@ -0,0 +1,478 @@
+from dataclasses import dataclass
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/pipeline/runtime.py:1]: Thanks for the review. We are not targeting Python 3.8/3.9 in this PR: `pyproject.toml` declares `requires-python = ">=3.11"`. Because of that, PEP 585 built-in generics and Python 3.10+ features such as `zip(..., strict=True)` are within the supported runtime, so we will keep this unchanged.

File: spikingjelly/activation_based/distributed/pipeline/spikformer.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
+from typing import Optional, Sequence, Union
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/pipeline/spikformer.py:1]: Thanks for the review. We are not targeting Python 3.8/3.9 in this PR: `pyproject.toml` declares `requires-python = ">=3.11"`. Because of that, PEP 585 built-in generics and Python 3.10+ features such as `zip(..., strict=True)` are within the supported runtime, so we will keep this unchanged.

File: spikingjelly/activation_based/distributed/tensor_parallel/linear.py [hunk 6]
```diff
@@ -0,0 +1,231 @@
+import inspect
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/tensor_parallel/linear.py:1]: Thanks for the review. We are not targeting Python 3.8/3.9 in this PR: `pyproject.toml` declares `requires-python = ">=3.11"`. Because of that, PEP 585 built-in generics and Python 3.10+ features such as `zip(..., strict=True)` are within the supported runtime, so we will keep this unchanged.

File: spikingjelly/activation_based/distributed/tensor_parallel/utils.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    target_children = list(target.named_children())
+    source_children = list(source.named_children())
+    if len(target_children) != len(source_children):
+        raise ValueError(
+            f"Cannot overwrite {type(target)} with {type(source)} because child counts differ: "
+            f"{len(target_children)} vs {len(source_children)}."
+        )
+    for (target_name, _), (_, source_child) in zip(
+        target_children, source_children, strict=True
+    ):
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/tensor_parallel/utils.py:29]: Thanks for the review. We are not targeting Python 3.8/3.9 in this PR: `pyproject.toml` declares `requires-python = ">=3.11"`. Because of that, PEP 585 built-in generics and Python 3.10+ features such as `zip(..., strict=True)` are within the supported runtime, so we will keep this unchanged.

File: spikingjelly/activation_based/distributed/tensor_parallel/channel.py [hunk 8]
```diff
@@ -0,0 +1,581 @@
...
+    if isinstance(padding, str):
+        if padding == "valid":
+            return tuple(0 for _ in range(2 * len(source.kernel_size)))
+        if padding != "same":
+            raise ValueError(f"Unsupported padding string '{padding}'.")
+        total_padding = [
+            dilation * (kernel - 1)
+            for dilation, kernel in zip(
+                source.dilation, source.kernel_size, strict=True
+            )
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/tensor_parallel/channel.py:30]: Thanks for the review. We are not targeting Python 3.8/3.9 in this PR: `pyproject.toml` declares `requires-python = ">=3.11"`. Because of that, PEP 585 built-in generics and Python 3.10+ features such as `zip(..., strict=True)` are within the supported runtime, so we will keep this unchanged.

**Comment by AllenYolk** [comment]
@coderabbitai review

**Comment by AllenYolk** [comment]
@gemini-code-assist review

**Comment by AllenYolk** [comment]
@coderabbitai review

**Comment by AllenYolk** [comment]
@gemini-code-assist review

**Comment by AllenYolk** [comment]
@gemini-code-assist review

**Comment by AllenYolk** [comment]
@coderabbitai review

File: spikingjelly/activation_based/distributed/tensor_parallel/channel.py [hunk 9]
```diff
@@ -0,0 +1,581 @@
...
+
+    padding = source.padding
+    if isinstance(padding, str):
+        if padding == "valid":
+            return tuple(0 for _ in range(2 * len(source.kernel_size)))
+        if padding != "same":
+            raise ValueError(f"Unsupported padding string '{padding}'.")
+        total_padding = [
+            dilation * (kernel - 1)
+            for dilation, kernel in zip(
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/tensor_parallel/channel.py:28]: Thanks. This project declares Python >= 3.11, so Python 3.8/3.9 compatibility is outside the supported baseline. We keep `zip(..., strict=True)` here because it documents and enforces the invariant.

File: spikingjelly/activation_based/distributed/tensor_parallel/utils.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/tensor_parallel/utils.py:29]: Thanks. This project declares Python >= 3.11, so Python 3.8/3.9 compatibility is outside the supported baseline. Although the lengths are checked just above, keeping `zip(..., strict=True)` is intentional as an executable invariant.

**Comment by AllenYolk** [comment]
@coderabbitai review

**Comment by AllenYolk** [comment]
@gemini-code-assist review

**Comment by AllenYolk** [comment]
@gemini-code-assist review

**Comment by AllenYolk** [comment]
@coderabbitai review

File: spikingjelly/activation_based/distributed/tensor_parallel/utils.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/tensor_parallel/utils.py:29]: We intentionally keep `strict=True` here. This PR targets the repository declared Python baseline of Python >= 3.11, so Python 3.8/3.9 compatibility is out of scope for this change.

File: spikingjelly/activation_based/distributed/tensor_parallel/channel.py [hunk 10]
```diff
@@ -0,0 +1,622 @@
...
+                source.dilation, source.kernel_size, strict=True
+            )
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/tensor_parallel/channel.py:30]: We intentionally keep `strict=True` here. This PR targets the repository declared Python baseline of Python >= 3.11, so Python 3.8/3.9 compatibility is out of scope for this change.

**Comment by AllenYolk** [comment]
@gemini-code-assist review

**Comment by AllenYolk** [comment]
@coderabbitai review

File: spikingjelly/activation_based/distributed/tensor_parallel/channel.py [hunk 11]
```diff
@@ -0,0 +1,622 @@
...
+        ]
```
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/tensor_parallel/channel.py:31]: This is intentionally left as `strict=True`. The repository now declares Python >= 3.11 as the supported baseline, so Python 3.8/3.9 compatibility is out of scope for this PR.

File: spikingjelly/activation_based/distributed/tensor_parallel/utils.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (AllenYolk) [reply spikingjelly/activation_based/distributed/tensor_parallel/utils.py:29]: This is intentionally left as `strict=True`. The repository now declares Python >= 3.11 as the supported baseline, so Python 3.8/3.9 compatibility is out of scope for this PR.

**Comment by AllenYolk** [comment]
@coderabbitai review

**Comment by AllenYolk** [comment]
@gemini-code-assist review

**Comment by AllenYolk** [comment]
@coderabbitai review

**Comment by AllenYolk** [comment]
@gemini-code-assist review

**Comment by AllenYolk** [comment]
@coderabbitai review

**Comment by AllenYolk** [comment]
@gemini-code-assist review
